core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -304,15 +308,18 @@ public void sendMessage(ReqT message) {
     Preconditions.checkState(stream != null, "Not started");
     Preconditions.checkState(!cancelCalled, "call was cancelled");
     Preconditions.checkState(!halfCloseCalled, "call was half-closed");
-    boolean failed = true;
+    Status failure = Status.CANCELLED;
     try {
       InputStream messageIs = method.streamRequest(message);
       stream.writeMessage(messageIs);
-      failed = false;
+      failure = null;
+    } catch (RuntimeException e) {
```

Comment:
This doesn't catch `Error`, which would give the user a vague `CANCELLED` status without details.
I would `catch (Throwable e)`, and use `Throwables.propagate(e)`.

I am not sure whether we should rethrow the exception, given that it's already propagated to the status. It feels redundant. @ejona86 @nmittler WDYT?